src/main/java/com/example/demo/repository/SavedEmails.java

```diff
@@ -0,0 +1,70 @@
+package com.example.demo.repository;
+
+import com.example.demo.consoleInfo.Email;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.stereotype.Component;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import java.util.List;
+
+@Component
+public class SavedEmails implements EmailRepo {
+    private final static String fileForStorage = "listOfEmails.json";
+    private final ObjectMapper mapper;
+    public SavedEmails(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+    @Override
+    public void save(List<Email> email) throws IOException {
+        try (PrintWriter out = new PrintWriter(new BufferedWriter(
+                new FileWriter(fileForStorage, true)))) {
+            mapper.writeValue(out,email);
+        }
+    }
+    @Override
+    public List<Email> findAllEmails() throws IOException {
+        return mapper.readValue(new File(fileForStorage), new TypeReference<>() {
+        });
+    }
+
+    @Override
+
+        public List<Email> showEmailsForSending() throws IOException {
+            LocalDateTime currentTime = LocalDateTime.now();
+            List<Email> emailList = findAllEmails();
+            List<Email> result = new ArrayList<>();
+
+            for (Email email : emailList) {
+                if (currentTime.equals(email.getDate())) {
+
+                    result.add(email);
+                }
+            }
+            return result;
+        }
+
+        @Override
+        public void deleteSentEmails() throws IOException {
+            LocalDateTime currentTime = LocalDateTime.now();
+            List<Email> emailList = findAllEmails();
+            List<Email> resultList = new ArrayList<>();
+
+            for (Email email : emailList) {
+                if (!currentTime.equals(email.getDate())) {
```

Comment:
It's better to rely on additional property - sent. Because, think about a situation when you stored emails for delivery for tomorrow but executed this program one day after tomorrow. All emails will be gone even though there were not  sent. To make it simpler, you can make this method take one argument(or a list of emails) that should be deleted, and execute this method in component responsible for delivery